.github/workflows/cygwin.yml

```diff
@@ -0,0 +1,82 @@
+name: Test on Cygwin
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+jobs:
+  cygwin_build_test:
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+          fetch-depth: 0
+      - name: Install Cygwin
+        uses: egor-tensin/setup-cygwin@v3
+        with:
+          platform: x64
+          install-dir: 'C:\tools\cygwin'
+          packages: >
+            python37-devel python37-pytest python37-hypothesis
+            python37-cython python37-pip python37-wheel
+            python37-setuptools liblapack-devel libopenblas
+            gcc-fortran git dash
+      - name: Set Windows PATH
+        uses: egor-tensin/cleanup-path@v1
+        with:
+          dirs: 'C:\tools\cygwin\bin'
+      - name: Verify that bash is Cygwin bash
+        run: |
+          command bash
+          bash -c "uname -svrmo"
+      - name: Update with Cygwin git
+        # fetch-depth=0 above should make this short.
+        run: |
+          dash -c "which git; /usr/bin/git fetch --all -p"
+      - name: Verify python version
+        # Make sure it's the Cygwin one, not a Windows one
+        run: |
+          dash -c "which python3.7; /usr/bin/python3.7 --version -V"
+      - name: Build NumPy wheel
+        run: |
+          dash -c "/usr/bin/python3.7 setup.py bdist_wheel"
+      - name: Install new NumPy
+        run: |
+          bash -c "/usr/bin/python3.7 -m pip install dist/numpy-*cp37*.whl"
+      - name: Check that pip installed the extensions
+        run: |
+          dash -c "/usr/bin/python3.7 -m pip show numpy"
+          dash -c "/usr/bin/python3.7 -m pip show -f numpy | grep .dll"
+          echo >list_dlls_dos.sh 'site_packages=$(python3.7 -m pip show numpy | \
+              grep Location | cut -d " " -f 2 -);
+          dll_list=$(for name in $(python3.7 -m pip show -f numpy | \
+              grep -F .dll); do echo ${site_packages}/${name}; done)
+          ls -l ${dll_list}
+          file ${dll_list}
+          ldd ${dll_list}
+          cygcheck ${dll_list}
+          cd dist/
+          for name in ${dll_list};
+          do
+              echo ${name}
+              python3.7 -c "import "$(echo ${name} | \
+                  sed -E \
+                  -e "s/\/+(home|usr).*?site-packages\/+//g" -e "s/\//./g" \
+                  -e "s/.cpython-3.m?-x86(_64)?-cygwin.dll//g")
+          done
+          '
```

Comment:
Stray?

Reply:
I was wondering about `'`.

Reply:
But now I see it starts way up above :)

Reply:
Yes, that's the end of the extension-module-checking script.  Should I split that into a separate file somewhere rather than writing it out each time?

Reply:
I think the first thing is to get the test passing again, we can worry about further cleanups after that. I'm beginning to regret the original suggestion :) But it does look _much_ nicer now.